### DISCUSSION TITLE
Option to recreate the pods after externally managed secrets change

### DISCUSSION THREAD

**Discussion by JasminGruschke** [concourse/concourse#8440] [Help & Support] [answered]
Hi there,
I manage the secrets for web/workers "externally", i.e. create the k8s secrets independently of the concourse helm chart. However, rotating secrets (like certificates or passwords), the "old" secrets are still mounted within the web/worker pods.
Is there an option to hint concourse to recreate the pods in order to propagate the secrets / remount the secrets?
That'd be great!

**Comment by lrstanley** [comment, marked as answer]
Take a look at [Reloader](https://github.com/stakater/Reloader), it works through an annotation, and it will watch any configmaps and secrets, and automatically allow the pod to get recreated when items change.

**Reply by JasminGruschke** [reply]
Thx a lot - yet I hoped for something directly in concourse ;-)

**Reply by lrstanley** [reply]
Understandable, however given there are so many different configuration options, it's not very feasible for a project to use mounted secrets/configmaps for all configuration options. If it were an option, I'd suspect it'd only be a small handful of options that could dynamically change based off the mounted value changing. It also introduces a very larger overhead internally to support such options, as you'd have to close/reopen connections, stop/restart various types of internal tasks, and add additional forms of mutexing.

This is why the vast majority of k8s based deployments don't support such a thing, unless they're specifically operators. I.e. reloader is the common go-to if you're looking for this type of functionality.

**Reply by JasminGruschke** [reply]
And that's certainly why it has so many stars 🥇 

thx a lot @lrstanley
